packages/checkbox/theme/lumo/vaadin-checkbox-styles.js

```diff
@@ -52,31 +53,28 @@ registerStyles(
 
     /* Checkmark */
     [part='checkbox']::after {
-      content: '';
       pointer-events: none;
       display: inline-block;
-      width: 0;
-      height: 0;
-      border: 0 solid var(--lumo-primary-contrast-color);
-      border-width: 0.1875em 0 0 0.1875em;
-      box-sizing: border-box;
-      transform-origin: 0 0;
       position: absolute;
-      top: 0.8125em;
-      left: 0.5em;
-      transform: scale(0.55) rotate(-135deg);
       opacity: 0;
+      contain: content;
     }
 
     :host([checked]) [part='checkbox']::after {
```

Comment:
Doesn't scale nicely with font-size

![Screenshot 2022-01-17 at 17 24 00](https://user-images.githubusercontent.com/1222264/149796929-d5c7ffd1-0e05-4f74-8b70-cd8ffa2c78c4.png)

Reply:
It shouldn't scale with the font size. Thanks for the reminder. The size of the icon should be based on the sizing properties.

Reply:
Could you please elaborate a bit more, which sizing properties do we need to set for the icon to look properly?

Reply:
> It shouldn't scale with the font size.

Oh, I assumed so because the [V14 checkbox](https://vaadin.com/docs/v14/ds/components/checkbox) does. The [issue](https://github.com/vaadin/web-components/issues/3262) this PR fixes is also about font-size.

Reply:
Originally I was using `--lumo-font-size-l` defined with `rem` (that's why I thought it would be a better fit).
@jouni recommended to replace it with `--lumo-icon-size-s` which uses `em` and does not scale.

Should we define fixed font-size on the checkbox part to mitigate this issue?

Reply:
I opened #3309 to discuss the border radius issue.

Reply:
I agree that it makes sense to use `rem` for border-radius. Let's fix that separately.
In the meanwhile, are there any other changes needed to proceed with this PR?

Reply:
I pushed an update.